### DISCUSSION TITLE
Synology NAS

### DISCUSSION THREAD

**Discussion by olivierLA75** [guydavis/machinaris#233] [Q&A] [answered]
Hi,
I have a Windows 10 workstation where Machinaris is installed and I have a Synology NAS that has four different volumes that are mapped to the computer as network hard drives. I did put the four network drives with their respectives letters and folders in the launch wizard but it seems that Machinaris is not counting the plots on these. is it normal? is there a better way to do this? should I install docker on the Synology? not sure what to do. thanks
PS: I have attached here the windows screenshot that shows the 4 volumes
<img width="767" alt="Screen Shot 2021-08-18 at 6 40 22 PM" src="https://user-images.githubusercontent.com/83734065/129994103-b0839fa6-9373-41c6-8a70-18ffc5e03e6d.png">

**Comment by guydavis** [comment]
Yes, this is a known issue related to how Docker on Windows accesses (or fails to) network shares.  The Docker run command succeeds, the folder is appears inside the container, but the folder is completely empty (inside).  So from the point of view of Machinaris, the folder contains no plots.  Again, to emphasize, this is an issue with Docker on Windows and affects any Docker app trying to access network shares.

First, please try this [workaround](https://www.reddit.com/r/sonarr/comments/nk8nij/windows_docker_wsl2_unable_to_mount_network_drive/gzc5xhd?utm_source=share&utm_medium=web2x&context=3) suggested by a Reddit user with a Synology NAS he was trying to get Radarr (in Docker) to access his movie files.  Please let me know if this works in your case.

Alternatively, you could try running Machinaris as a harvester directly on your NAS, if it allows Docker containers to run.  This would then access the plots directly on the NAS to farm them.  In that case, the Windows harvester would only farm plots on directly attached drives.  

Hopefully one of these two options works for you.

**Comment by olivierLA75** [comment]
the second option works for me (harvester on NAS)
<img width="1017" alt="Screen Shot 2021-08-18 at 8 18 18 PM" src="https://user-images.githubusercontent.com/83734065/130002324-2c22f408-0a51-466e-8b58-4710f8428b49.png">
<img width="1017" alt="Screen Shot 2021-08-18 at 8 18 29 PM" src="https://user-images.githubusercontent.com/83734065/130002332-76188d67-1d1f-439f-8faf-1b94db46059c.png">
<img width="1020" alt="Screen Shot 2021-08-18 at 8 18 47 PM" src="https://user-images.githubusercontent.com/83734065/130002341-d0e59486-08fc-4d48-a70b-b32cbd174946.png">
<img width="1016" alt="Screen Shot 2021-08-18 at 8 18 54 PM" src="https://user-images.githubusercontent.com/83734065/130002350-d2b26517-85da-40e0-8b42-3b35c94dccfe.png">
. I've installed Machinaris on docker and it's running. I listed all the volumes but I'm not sure which command to launch on the terminal, none of them seems to work. I tried to browse to port 8926 but it's not working.

**Reply by guydavis** [reply]
mode=harvester
controller_host=IP_OF_YOUR_CONTROLLER

See, the Workers tab of your controller's Machinaris WebUI and click New Worker button for details.  While it's true you did launch Machinaris, without the environment variables above, it won't connect to the controller and function just as a harvester.  Use the New Worker wizard which shows a Docker run command and map it to the NAS Docker GUI you show above.

**Comment by olivierLA75** [comment]
will the network shares work on Linux? my other harvesters are on Linux

**Reply by guydavis** [reply]
Yes

**Comment by olivierLA75** [comment]
Is it better to have the farmer on Linux than Windows or is it the same?

**Reply by guydavis** [reply]
Up to your situation.  I use both.

**Comment by olivierLA75** [comment]
So I think I have figured out everything but I’m not sure about one thing. I used to farm using the windows GUI and several plotters and I just switched everything to Linux with Machinaris everywhere.

I have OG plots that I intend to keep, I have some pool plots with Space Pool and a few thousands portable plots for self pool. I am thinking about moving the latter to Space Pool if possible (not sure how to do that though)

I imported the 24 words in the farmer and didn’t add any other info, seems it figured out the Farmer public key and Pool public key. Now I’m starting to set up workers that all do harvester and plotter from the farmer’s Machinaris UI. It automatically filled up the farmer public key and pool public key info but the pool contract address is empty: do I need to fill that field up on the farmer? On the harvesters? And if yes where to find it? I’m just a little confused…

**Reply by guydavis** [reply]
Hi, once the Machinaris fullnode/controller is fully synced, you should see your chosen pool on the Settings | Pools page.  This is the same output as `chia plotnft show` at the CLI (in-container).  Do you see your PlotNFT details there including a `pool_contract_address`?  

Then go to Settings | Plotting and for each worker verify that you have uncommented `pool_contract_address` and commented out `pool_pk`.  Be sure to Save each worker's config.  The pool_contract_address should be automatically filled in if you have a PlotNFT.  Additional details on[ the wiki](https://github.com/guydavis/machinaris/wiki/Pooling#setup-and-config).

**Comment by olivierLA75** [comment, marked as answer]
Thanks Guy.
My controller/farmer isn’t fully syncd yet but it should be done soon so I will wait for that before adding the workers.

 So if I understood this right, you can only use one pool/NFT at a time with Machinaris (which is fine for me) and not a few like you can do on the Chia GUI. 

Ok so I will let the pool contract address blank/undocumented, but what is it for then? I saw that it was filled up in your wiki screenshots

**Reply by guydavis** [reply]
Once your controller is synced, any existing PlotNFTs you have created will be shown on the Pools page.  Machinaris does work with multiple pools, but configuration is done via the `chia plotnft` command-line as the current Settings | Pools page is meant for simple usage by those with only one pool config.  

Once synced, the `pool_contract_address` will auto-fill and you should review and Save for each worker on the Settings | Plotting.  In your case, you definitely want to set your preferred address as you seem to have multiple pools on the go at once.

**Comment by olivierLA75** [comment]
Noted thanks.
My farmer on Windows is almost synced but I realized that I had left the path of the network hard drives I need to remove. Can I just remove them in the settings/config.yaml? Or does it need to be done in the container? I would hate to have to re-sync the whole thing as it took forever

**Reply by guydavis** [reply]
No need to resync.  You shouldn't need to edit config.yaml by hand.  Just use `chia plots remove -d /OLD/IN-CONTAINER/PATH`.  Detail on the [wiki](https://github.com/guydavis/machinaris/wiki/FAQ#can-i-use-two-or-more-final-destination-directories-to-hold-my-plots).

**Comment by goldcd** [comment]
I plotted/farmed from my Windows PC with plots on my Synology - and latency became a problem (especially when trying to dump plots and farm over the same congested connection).

Switched to farming on the Synology, and it behaves much better.

Today I switched to Machinaris - and here comes my problem..

Installed fullnode following the guide - https://github.com/guydavis/machinaris/wiki/Synology
(and copied across my dbs to prevent a rebuild).

Chia itself seems perfectly happy - and is happily synced and farming away if I check via command line.

What doesn't seem to be being detected is the farming via the Machinaris UI.

My config is:
![image](https://user-images.githubusercontent.com/25711112/148610234-3c36c395-f3d3-4e54-a719-4b245464fac5.png)

But farmer doesn't seem to connect:
![image](https://user-images.githubusercontent.com/25711112/148610386-4ff88774-4c3d-4f19-8d46-e97de659a11c.png)

Problem seems to be the null address for the worker hostname. I'd thought maybe the issue was a missing host in the variables - but setting them doesn't seem to help (and I'd assumed they're to be used for harvester containers to call home)

**Reply by guydavis** [reply]
Hi @goldcd! Thanks for providing all that useful detail in your report.  Yes sorry for the complexity of installation/configuration on Synology, requires a lot of hand-inputs.  However, once configured is fairly stable.  In your case, please put the machine's IP (on your LAN) into `worker_address` instead of leaving it null as shown in your above screenshot.  Then restart the Machinaris container. Hope this helps!

**Reply by goldcd** [reply]
Thanks for the quick response - I gave it a go - and it worked!
..not that I doubted you, but :)

Initially didn't seem to work - a new farmer/worker on the IP address appeared, but wasn't initially responding (gave same error as the null) - but after a few minutes it seemed to sort itself out and I could prune the broken  'null' worker and everything looks good.

![image](https://user-images.githubusercontent.com/25711112/148617718-4ab12572-4b49-463b-b6c5-0c00d3fb2d7b.png)

#Edit - I've just realized you'd told me to do that in the guide I linked
*patiently waits for earth to swallow me*
